lib/gauge_field.cpp

```diff
@@ -274,12 +888,234 @@ namespace quda {
     }
   }
 
-  std::ostream& operator<<(std::ostream& output, const GaugeFieldParam& param) {
+  void *create_gauge_buffer(size_t bytes, QudaGaugeFieldOrder order, QudaFieldGeometry geometry)
```

Comment:
Should there be a CUDA/CPU toggle here to change between allocating device vs host buffers? (Same question for the next few functions as well.)

Reply:
Nope, at least not in the present form.  These buffers are only used for device allocation.  I eventually plan to clean this stuff up, but in the spirit of the internals of `GaugeField` be similar to what was in `cuda`/`cpu` fields prior, this is deferred.